cardano-api/src/Cardano/Api/Tx.hs

```diff
@@ -433,13 +432,15 @@ getTxBody (ShelleyTx era tx) =
                     (Map.elems msigWits)
                     TxBodyNoScriptData
                     (strictMaybeToMaybe txAuxiliaryData)
+                    TxBodyScriptValidityNone
 
     getAlonzoTxBody :: forall ledgerera.
                        ShelleyLedgerEra era ~ ledgerera
                     => ScriptDataSupportedInEra era
+                    -> TxBodyScriptValiditySupportedInEra era
```

Comment:
We don't need to parameterize this. We can hardcode the value on line 459 as we know we are in the Alonzo era.

Reply:
I think it's still needed.  If we hardcode the value, we still have to prove the hardcoded value matches the `era` parameter.  Unless we deparameterize the entire function.

For example `ScriptDataSupportedInEra era` argument pretty much does the same thing.